### ISSUE TITLE
Entire torrent being downloaded regardless of what files from it I have chosen

### ISSUE THREAD

**Issue by vi** [Tribler/tribler#2566]
##### Tribler version/branch+revision:

1f65ea44c14dc904efb222a5ae4a616489a52475 devel
##### Expected behavior:

When I choose not all files from a torrent, Tribler tries to economize traffic and storage space and does not request non-relevant files. Size of `.unwanted` directory is insignificant.
##### Actual behavior:

Entire torrent being downloaded and placed to `.unwanted`. Checkboxes only affects placement of the file inside or outside of `.unwanted`.

It wasn't so before as far as I remember.

**Comment by synctext** [comment]
@vi thnx for testing our bleeding edge code. Coming weeks we'll try to get Tribler7 our of the door, expect more bugs like this..

**Comment by vi** [comment]
@synctext , Although it works on stable, it is not a recent devel breackage.

I also can't easily go back to "stable" or "next" because of database incompatibilities.

**Comment by vi** [comment]
I also tried testing the new GUI pull request, but it still doesn't work for me (unless it's `START_FAKE_API`) and the developer stated that fixing it is [postponed by something else](https://github.com/devos50/tribler/issues/10#issuecomment-248229242)

**Comment by devos50** [comment]
@vi I recently continued work on Tribler after graduation and a holiday. There are some things left regarding the interface design, REST API and Tribler core for the Tribler 7 release which I'm currently working on.

A long outstanding point of attention was to make the unit tests of Tribler stable to improve developer productivity. I managed to do this for most part in PR #2555. Now, I am working on a refactoring of the configuration system of Tribler (which is complex and contains various unnecessary settings) in PR #2564. Doing this allows me to enhance the REST API with an endpoint to dynamically update Tribler settings. In parallel, I will finish the design of the new GUI and finish the Windows/MacOS/Linux packagers and unit tests.

After the configuration system has been refactored, I will improve the downloading mechanism during which I can take a look at this issue.

Thank you for testing Tribler! Please keep track of #2153 to be updated about the progress of the new user interface.

**Comment by vi** [comment]
I've juts tried it again and it still doesn't work. Now I'm running legacy GUI `./tribler.sh` from Tribler commit 05328974eb81ed06eafa610ef2801b1f58ff09b1 with one rebased commit on top. There it also downloads files regarless of their included/excluded status.

**Comment by vi** [comment]
Asserting that in 97f2766c46f2728c91d6a48f4925ae3109786d84 (v6.5.2) it downloads the files as it should.

**Comment by synctext** [comment]
@vi Things a bit slow now, lot of new developers and new projects starting: https://github.com/Tribler/tribler/wiki#current-items-under-active-development

**Comment by devos50** [comment]
@vi I verified your report: all files are being downloaded. I will investigate it and this should be fixed before the Tribler 7 (public alpha) release.

**Comment by devos50** [comment]
Well, this bug has been introduced when the credit mining functionality was integrated in Tribler. It will be fixed with #2691

**Comment by vi** [comment]
Verified that in d12a8515b4f3fd976030b2bb04e2796dd5788499 the bug is not reproducible.
